src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -109,6 +118,41 @@ public virtual bool CanValidateToken
             get { return true; }
         }
 
+        private JObject CreateDefaultJWEHeader(EncryptingCredentials encryptingCredentials, string compressionAlgorithm)
+        {
+            var header = new JObject();
+
+            if (!string.IsNullOrEmpty(encryptingCredentials.Alg))
```

Comment:
Alg, Enc can not be null.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b16a4d5c90bef3a20b35282ef4dc1fb1b4d8cd49/src/Microsoft.IdentityModel.Tokens/EncryptingCredentials.cs#L100

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b16a4d5c90bef3a20b35282ef4dc1fb1b4d8cd49/src/Microsoft.IdentityModel.Tokens/EncryptingCredentials.cs#L109

Reply:
Good point! Fixed.